section3/reflection.md

```diff
@@ -2,12 +2,27 @@
 
+  * I found this section a little more challenging than the last sections. I found most of my problems when working on the loops-arrays exercise. I was able to solve most of my questions within a reasonable time frame. There were a couple questions that I would get stuck on, and normally if I moved forward or took a quick break, and changed my mindset, I was able to quickly find an answer. The one that gave me the most trouble was doubling the array. I was able to find the method I needed when I looked back at the method notes I took.  
+
 2. Think of a web or mobile application you use regularly. Where is an example of a place that the developers probably use Arrays? Explain.
 
+  * I was on the Chipotle app the other night and I think the menu where you order your ingredients is an example of an array. They have different variable titles like rice or protein and under them is a list of all the ingredients that fall within that variable. So to me these could be arrays.
+
+
 3. In your own words, explain what an Array method is.
 
+  * An array method is our way of changing or modifying an array. If we piggy back off the prior Chipotle example. You can use methods to add different ingredients to each of the arrays, you can also subtract items if they run out. You can sort the array to make it easier to navigate for the user. So we can use methods to change an array in many different ways.  
+
+
 4. What is the purpose of a loop? When we would use a loop in conjunction with array vs. without?
 
+  * The purpose of a loop is to repeat things. If you want to have something run a certain number of times you would use a loop. Loops are what make computers so capable and are a powerful tool when programming. If you wanted to have the computer count to a hundred you would use a loop without an array.You would want to use an array and a loop if you wanted to have multiple sets of data. For example, a shopping list where you want to add multiple items.
+
+
 5. If you had to teach a 5 year old about loops, how would you explain it?
 
+  * A loop is a lot like the game musical chairs. For the very first loop everyone lines up in a circle around the chairs. If there are 10 students there will be 9 chairs. When the first round of music plays everyone starts walking around the chairs. When the music stops everyone tries to pick a chair but one student is out. Since there are now 9 students the teacher removes a chair. The process is repeated until there are two students and 1 chair left. After this, the last student sits in a chair and the game or loop is over.  
+
 6. What questions do you still have about Arrays and loops?
+
+  * I still have a question about when you console log text with an array. When you are adding a string and an array together do you have to use the "+" ? or do commas do the same thing?
```

Comment:
Chances are you will iterate through the array and log the items in the array as you loop through it.